code-challenges.test.js

```diff
@@ -14,39 +16,238 @@
+//     expect(codeWords(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+//     expect(codeWords(secretCodeWord3)).toEqual("3cc3ntr1c")
+//  })
+// })
+
+//RED-Good Failure
+// FAIL  ./code-challenges.test.js
+// codeWords
+//   ✕ creates a function that takes a string and returns a string with a coded message
+
+// ● codeWords › creates a function that takes a string and returns a string with a coded message
+
+//   ReferenceError: codeWords is not defined
+
+//     25 | describe("codeWords", () => {
+//     26 |     it("creates a function that takes a string and returns a string with a coded message", () => {
+//   > 27 |     expect(codeWords(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+//        |     ^
+//     28 |     expect(codeWords(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+//     29 |     expect(codeWords(secretCodeWord3)).toEqual("3cc3ntr1c")
+//     30 |  })
+
+//     at Object.expect (code-challenges.test.js:27:5)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.15 s
+// Ran all test suites.
+
 
 // b) Create the function that makes the test pass.
 
+//PseudoCode
+//Input: secretCodeWords1, 2, 3
+//Output: "L4ck4d41s1c4l", "G0bbl3dyg00k", "3cc3ntr1c"
+//Process:
+
+//:codeWords, takes in a string as a parameter
+//create a new object that holds key value pairs for each letter to number, implemented later in the function
+// initially tried to create a new variable after the object to store a string turned into an array by using .split
+// then created a line that .map the new variable after it was split
+//then created a for loop to iterate through the array
+//created an if/else conditional statement saying 
+    //if (newArray = "a", "e", "i", "o", "u")
+        //return string.replaceAll("a", "4")
+
+//From here, did not know how to put in multiple values for .replaceAll("a", "4"). This is what's seen in the output. Also, turned it back into a string using .join before. Internet resources led me to the code below.
+
+//other code: created a new variable for the string. 
+//used a for loop to call on key in object
+//used the new variable to replaceAll with the key value from newCharacters, tried variations of writing what goes in the parathensis
+//return newString
+//Console log the original function and pass in the first arguement
+
+//First OUTPUT: L4ck4d4isic4l
+//Second OUTPUT: UNDEFINED at 'vaule' : newString = newString.replaceAll(newCharacters[key], newCharacters[value])
+
+
+// const secretCodeWord1 = "Lackadaisical"
+
+// const codeWords = (string) => {
+//     const newCharacters = {
+//         "a": "4",
+//         "e": "3",
+//         "i": "1",
+//         "o": "0",
+//         "E": "3"
+//     };
+//         let newString = string;
+//         for (let key in newCharacters) {
+//         newString = newString.replaceAll(newCharacters[key], newCharacters[value])
+//         return newString 
```

Comment:
I see the thought process, great effort! I like the idea of having an object to hold your vowels with their replacement letters. The issue here is that `value` is not defined anywhere. I assume you are trying to tell `.replaceAll` to replace the match with the value at that key. However, you used a `for in` loop, and you defined `key` as your parameter. So key is defined, but value is not. To fix this, you can do:
`newString = newString.replaceAll(key, newCharacters[key])`. Then, outside of your for in loop, you would return newString.